.gitignore

```diff
@@ -1,3 +1,4 @@
+composer.lock
```

Comment:
Why should `composer.lock` be ignored?

Reply:
A potential issue with `composer.lock` being committed in libraries is that it may capture unintentional dependencies, cascaded down from the PHP version, that was used for the file generation. The composer documentation says that it is [not necessary to keep a `composer.lock` around](https://getcomposer.org/doc/01-basic-usage.md#updating-dependencies-to-their-latest-versions) in libraries and infrastructure differences may be one of the reasons for this. Some developers even believe that this should be considered undesirable practice.
In contrast, versioning `composer.lock` for private projects is advisable and almost always a good practice. This is because for private projects there is better control over the infrastructure that runs the code.